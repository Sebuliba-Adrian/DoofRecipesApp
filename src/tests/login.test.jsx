import React from 'react';
import sinon from 'sinon';
import { mount, configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import LoginPage from '../components/authentication/login';
configure({ adapter: new Adapter() });

const props = {
  location: {
    data: {
      message: 'Test message'
    }
  },
  history: {
    replace: jest.fn()
  }
};

global.localStorage = {
  setItem: () => {},
  getItem: () => {}
};

describe('<LoginPage />', () => {
  const preventDefault = jest.fn(); //PreventDefault spy

  it('renders', () => {
    const tree = renderer.create(<LoginPage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has initial state', () => {
    const wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper.state().username).toEqual('');
  });

  it('submits the form', () => {

    const wrapper = mount(<LoginPage {...props} />);
    console.log(wrapper.instance());

    const userName = wrapper.find('#username');
    userName.simulate('change', {
      target: { name: 'username', value: 'testusername' }
    });
    userName.value = 'testusername';
    expect(userName.value).toEqual('testusername');

    const userPassword = wrapper.find('#password');
    userPassword.simulate('change', {
      target: { name: 'password', value: 'testpassword' }
    });
    userPassword.value = 'testpassword';
    expect(userPassword.value).toEqual('testpassword');

    const form = wrapper.find('form');
    form.find('button').simulate('submit', { preventDefault });
    expect(preventDefault).toBeCalled();
  });
});
