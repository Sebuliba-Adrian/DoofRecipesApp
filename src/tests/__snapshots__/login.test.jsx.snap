// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders three <Foo /> components 1`] = `
<div
  className="col-md-4 offset-md-4 col-xs-10 offset-xs-2"
>
  <div
    className="card mt-5 p-4"
  >
    <div
      className="card-block"
    >
      <div>
        <h3
          className="logo text-center mb-3"
        >
          Doof Recipes App
        </h3>
        <p
          className="grey-text text-center mb-5"
        >
          Record your favourite recipes 
        </p>
      </div>
      <div
        className="line-with-text font-weight-bold mb-4"
      >
        <div
          className="line"
        />
        <div
          className="line-text"
        >
          LOGIN
        </div>
        <div
          className="line"
        />
      </div>
      
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control mb-1"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-control mb-2"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <button
          className="btn btn-primary btn-sm col-md-12"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
  <div
    className="card mt-2"
  >
    <div
      className="card-block"
    >
      <p
        className="card-text text-center message"
      >
        Don't have an account? 
        <a
          className="card-link"
          href="/registration"
          onClick={[Function]}
        >
          Register
        </a>
      </p>
    </div>
  </div>
  <div
    id="snackbar"
  />
</div>
`;
